### PR TITLE
Use cached TriggerHandler records in UnitTests

### PR DESCRIPTION
**Pull request by meena-sfdc**
1. We eliminated the DML that creates Trigger_Hander__c records in every single unit if they do not already exist and instead use the cached “default” collection to reduce the time needed to complete unit tests. This change reduces our overall MetaCI build time. These changes effect only tests running internally (not in customer org). 
2. In order to prevent an error, the order of the REL_Relationships_Con_TDTM class was updated to fire after the CON_ContactMerge_TDTM class during a Contact merge. Therefore, in TDTM_DefaultConfig class, the load order for CON_ContactMerge_TDTM has been updated to 2 and load order for REL_Relationships_Con_TDTM has been updated to 3.

# Critical Changes

# Changes
- In order to prevent an error, we updated the execution order of the REL_Relationships_Con_TDTM class to fire after the CON_ContactMerge_TDTM class during a Contact merge. Therefore, in the TDTM_DefaultConfig class, the load order for CON_ContactMerge_TDTM has been updated to 2 and load order for REL_Relationships_Con_TDTM has been updated to 3.

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by meena-sfdc** [comment]
**lurch: attach W-024174

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x8000000AVhIfAAL" target="_blank">W-024174</a>

**Comment by meena-sfdc** [comment]
Code changes have been applied and the PR is ready for review

File: src/classes/TDTM_Config_API_Test.cls [hunk 1]
```diff
@@ -118,4 +118,20 @@ private class TDTM_Config_API_Test {
...
+    /*******************************************************************************************************
+    * @description Insert a new Account and then verify that there are still zero records in the 
+        Trigger_Handler__c object
+    */
+    @IsTest
+    private static void testCachedRecordsCreateNoTriggerRecords() {
+        // Test the below scenario only if it is not customer org
+        // Create a account to trigger the TDTM Triggerhandler
+        if (!UTIL_Namespace.isCustomerOrg()) {
+            Account acct = new Account(Name='Test Account 876');
```
[the remark this answers is not in the document]
Comment (meena-sfdc) [reply src/classes/TDTM_Config_API_Test.cls]: I checked that util class, but didn't know that I can create one record in that, thanks for your suggestion!

[the remark this answers is not in the document]
Comment (meena-sfdc) [reply src/classes/TDTM_Config_API_Test.cls]: Code review changes have been applied and the PR is ready.

**Comment by meena-sfdc** [comment]
Code changes have been applied and the PR is ready.

**Pull request merged**